clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -4701,9 +4737,13 @@ EmitExtVectorElementExpr(const ExtVectorElementExpr *E) {
 
     // Store the vector to memory (because LValue wants an address).
     Address VecMem = CreateMemTemp(E->getBase()->getType());
+    // need to zero extend an hlsl boolean vector to store it back to memory
```

Comment:
Probably better to phrase this comment in terms of packed/non-packed boolean vectors given the abstraction in LangOpts